assets/js/atomic/blocks/product-elements/button/frontend.tsx

```diff
@@ -86,20 +124,108 @@ const productButtonSelectors = {
 			state,
 		}: {
 			context: Context;
-			// For now, let's just use any.
-			// eslint-disable-next-line @typescript-eslint/no-explicit-any
 			selectors: any;
 			state: State;
 		} ) => {
-			return selectors.woocommerce.isThereMoreThanOneItem( {
-				context,
-				state,
-			} );
+			return (
+				selectors.woocommerce.isThereMoreThanOneItem( {
+					context,
+					state,
+				} ) && context.woocommerce.displayViewCart
+			);
+		},
+		hasCartLoaded: ( { state }: { state: State } ) => {
+			return state.woocommerce.cart !== undefined;
+		},
+		numberOfItems: ( {
+			state,
+			context,
+		}: {
+			selecotrs: any;
+			state: State;
+			context: Context;
+		} ) => {
+			const product = getProductById(
+				state.woocommerce.cart,
+				context.woocommerce.productId
+			);
+
+			return product?.quantity || 0;
+		},
+		numberOfItemsPrev: ( {
+			state,
+			context,
+		}: {
+			selecotrs: any;
+			state: State;
+			context: Context;
+		} ) => {
+			const product = getProductById(
+				state.woocommerce.prevCart,
+				context.woocommerce.productId
+			);
+
+			return product?.quantity;
+		},
+
+		shouldAnimationStart: ( store: {
+			context: Context;
+			selectors: any;
+			state: State;
+		} ) => {
+			const { context, selectors } = store;
+
+			const isFreshLoad =
+				selectors.woocommerce.hasCartLoaded( store ) &&
+				context.woocommerce.initialNumberOfItems !==
+					selectors.woocommerce.numberOfItems( store ) &&
+				context.woocommerce.numberOfItems !==
+					selectors.woocommerce.numberOfItems( store );
+
+			const isFromCart =
+				selectors.woocommerce.numberOfItemsPrev( store ) !==
+					undefined &&
+				selectors.woocommerce.numberOfItemsPrev( store ) !==
+					selectors.woocommerce.numberOfItems( store ) &&
+				selectors.woocommerce.numberOfItems( store ) !==
+					context.woocommerce.numberOfItems;
```

Comment:
Nit: given that we are calling `numberOfItems()` several times, could we call it at the beginning of the function and save its value into a variable? Same with `numberOfItemsPrev()`. I feel it would make the code easier to read.